# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/android/nsWindow.cpp
# Commit: 5e275c31e9f4
# Full Hash: 5e275c31e9f42def8437d7c8a87cf4cf7aabf0c6
# Author: Hiroyuki Ikezoe <hikezoe.birchill@mozilla.com>
# Date: 2020-11-11 09:55:06
# Regressor Bug: 1674694
# File Overlap Count: 1
# Description:
#   Bug 1674694 - Change APZEventResult::mHandledByRootApzc to Maybe<APZHandledResult>. r=botond
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96331
# ==============================================================================

diff -r 279eb8cf0dcd -r 5e275c31e9f4 widget/android/nsWindow.cpp
--- a/widget/android/nsWindow.cpp	Sat Nov 07 03:43:53 2020 +0000
+++ b/widget/android/nsWindow.cpp	Wed Nov 11 00:33:20 2020 +0000
@@ -360,9 +360,10 @@
         WheelDeltaAdjustmentStrategy::eNone);
 
     APZEventResult result = controller->InputBridge()->ReceiveInputEvent(input);
-    int32_t ret = (result.mHandledByRootApzc == Some(true))
-                      ? INPUT_RESULT_HANDLED
-                      : INPUT_RESULT_HANDLED_CONTENT;
+    int32_t ret =
+        (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
+            ? INPUT_RESULT_HANDLED
+            : INPUT_RESULT_HANDLED_CONTENT;
 
     if (result.mStatus == nsEventStatus_eConsumeNoDefault) {
       return ret;
@@ -502,9 +503,10 @@
         nsWindow::GetEventTimeStamp(aTime), nsWindow::GetModifiers(aMetaState));
 
     APZEventResult result = controller->InputBridge()->ReceiveInputEvent(input);
-    int32_t ret = (result.mHandledByRootApzc == Some(true))
-                      ? INPUT_RESULT_HANDLED
-                      : INPUT_RESULT_HANDLED_CONTENT;
+    int32_t ret =
+        (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
+            ? INPUT_RESULT_HANDLED
+            : INPUT_RESULT_HANDLED_CONTENT;
 
     if (result.mStatus == nsEventStatus_eConsumeNoDefault) {
       return ret;
@@ -702,9 +704,10 @@
     }
 
     APZEventResult result = controller->InputBridge()->ReceiveInputEvent(input);
-    int32_t handled = (result.mHandledByRootApzc == Some(true))
-                          ? INPUT_RESULT_HANDLED
-                          : INPUT_RESULT_HANDLED_CONTENT;
+    int32_t handled =
+        (result.mHandledResult == Some(APZHandledResult::HandledByRoot))
+            ? INPUT_RESULT_HANDLED
+            : INPUT_RESULT_HANDLED_CONTENT;
 
     if (result.mStatus == nsEventStatus_eConsumeNoDefault) {
       if (returnResult) {
@@ -725,7 +728,7 @@
       return;
     }
 
-    if (result.mHandledByRootApzc != Nothing()) {
+    if (result.mHandledResult != Nothing()) {
       // We know conclusively that the root APZ handled this or not and
       // don't need to do any more work.
       switch (result.mStatus) {
